### ISSUE TITLE
Error: Bad packet length

### ISSUE THREAD

**Issue by elbertcastaneda** [mscdex/ssh2#273]
Uncaught Error: Bad packet length
[79121:0524/182004:ERROR:nw_shell.cc(335)] Error: Bad packet length
    at SSH2Stream._transform (/myApp/node_modules/ssh2/node_modules/ssh2-streams/lib/ssh.js:412:25)
    at SSH2Stream.Transform._read [as __read] (_stream_transform.js:167:10)
    at SSH2Stream._read (/myApp/node_modules/ssh2/node_modules/ssh2-streams/lib/ssh.js:213:15)
    at SSH2Stream.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:295:12)
    at writeOrBuffer (_stream_writable.js:282:5)
    at SSH2Stream.Writable.write (_stream_writable.js:210:11)
    at Socket.ondata (_stream_readable.js:524:20)
    at emitOne (events.js:80:20)
    at Socket.emit (events.js:150:7)

**Comment by mscdex** [comment]
Can you provide some kind of instructions on how I can duplicate the error (e.g. is there a docker container or something similar that is being used)?

**Comment by elbertcastaneda** [comment]
yes :

```javascript
'use strict';

var Client = require('ssh2'),
    net = require('net'),
    login2Server = function (user, password) {
        var self = this,
            status = 'd',
            sshSock = net.createConnection(22, '0.0.0.0'),
            connection = new Client();

        console.log('Connecting to server ....');

        connection.connect({
            sock: sshSock,
            username: user,
            password: password
        });

        connection.on('error', function () {
            if (status === 'c') {
                connection.end();
                connection.destroy();
            }
        });

        connection.on('connect', function () {
            console.log('ssh2 connect');
        });

        connection.on('ready', function () {
            //Never enter here
            console.log('ssh ready');
            status = 'c';

        });

        self.connection.on('end', function () {
            console.log('ssh end');
            status = 'd';
            console.log('end');
        });

        self.connection.on('close', function () {
            status = 'd';
            console.log('Connection :: close');
        });
    };

login2Server('root', '1234');

```

This error is showing only the new versions, because the error It was seen after updating my project with npm update (I think it's the library ssh2 -stream), the version without error is this:

{
"name": "ssh2",
  "version": "0.4.6",
  "author": {
    "name": "Brian White",
    "email": "mscdex@mscdex.net"
  },
  "description": "SSH2 client and server modules written in pure JavaScript for node.js",
  "main": "./lib/client",
  "engines": {
    "node": ">=0.8.7"
  },
  "dependencies": {
    "readable-stream": "~1.0.0",
    "ssh2-streams": "~0.0.5"
  }
}

**Comment by mscdex** [comment]
This should be fixed now in cec7b061c2. Please try it and let me know if it works for you.
